File: 1. cpp/1. basics/lecture 4/reverse number.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
        int reversed = 0;
        while (n != 0) {
            int digit = n % 10;
            reversed = reversed * 10 + digit;
            n /= 10;
        }
        cout << reversed << endl;
    }
};

/* output:



*/

int main() {
  int n;
  cout<<"Enter number:"<<endl;
  cin>>n;

  cout<<"output:"<<endl;
  solution pattern(n);

}

File: 2. python/1. basics/lecture 4/reverse number.py
class solution:
  def __init__(self, n):
    reversed =0
    while n>0:
      digit = n % 10
      reversed = reversed * 10 + digit
      n = n // 10
    print(reversed)

if __name__ == "__main__":
  n = int(input("Enter a number: "))
  obj = solution(n)

    